common/djangoapps/third_party_auth/strategy.py

```diff
@@ -41,6 +42,15 @@ def setting(self, name, default=None, backend=None):
                 if error_url:
                     return error_url
 
+        # Special case: we want to get this particular setting directly from the provider database
+        # entry if possible; if we don't have the information, fall back to the default behavior.
```

Comment:
The docstring above is incorrect for SAML providers. The SAML backend checks the global SAML SP config, not the ProviderConfig.